tgui/packages/tgui/interfaces/Wizard_Spellbook.js

```diff
@@ -105,32 +105,54 @@ const Spell = (props, context) => {
   const { spell, category } = props;
 
   const [purchased, setPurchased] = useSharedState(context, spell+"p", false);
-  let spell_contents = []; // Non-associated list of: desc, cost, cooldown, vr_allowed
+  let spell_contents = []; // Non-associated list of: desc, cost, cooldown, vr_allowed, spell_img
   for (let spell_data in spellbook_contents[category][spell]) {
     spell_contents.push(spellbook_contents[category][spell][spell_data]);
   }
 
   return (
     <Stack.Item>
-      <Section>
+      <Section mt={0.5} mb={-1.5}>
         {(vr === 1) && (spell_contents[3] === 0) && (
-          <Dimmer>
+          <Dimmer mb={3} mt={-2}>
             <Box fontSize={1.5} backgroundColor={"#384e68"} p={2}>
               Spell unavailable in VR
             </Box>
           </Dimmer>
         )}
         <Section
-          title={spell}
-          buttons={
-            <Flex>
-              <Button
-                backgroundColor={"green"}
-                disabled={spell_slots < spell_contents[1] || purchased}
-                onClick={() => { setPurchased(true); act("buyspell", { spell: spell }); }}
-              >
-                {If_Purchased_Text(purchased, spell_contents[1], spell_slots)}
-              </Button>
+          title={
+            <Flex justify={"space-between"} align={"end"}>
+              <Flex.Item>
+                <Flex>
+                  <Flex.Item>
+                    {!!spell_contents[4] && (
```

Comment:
```suggestion
                    {!!spell_contents[4] && (
```
This is definitely getting too unwieldy for a tuple of data and should be migrated to being an object with key/value pairs.

Reply:
I would love that, I had tried a lot mind you at the very start of the creation of the spellbook TGUI and I even had a help of a dev (Leah) and a nother person (Dr. Romayne) from the discord, yet we couldnt find a workable solution that didnt cause errors, so instead we did Dr. Romaynes solution that I think they used for their TGUIification of the pump control computer where they had the same problem.

Altough if this doesnt require a re-making of `spellbook_contents` and there is instead a simpler solution that can be fitted just for `spell_contents` I would greatly appreciate that, since I **could** try, but I doubt I have the expertise to make something that doesnt look even uglier.

Reply:
Curious what kind of errors you were seeing! This should be a pretty straightforward and error-free migration. I'm okay with it being done as follow-up work, I suppose. 5 fields that have to happen to be in there in a certain order does not make for maintainable code.